src/app/shared/dovetail/dovetail-audio.ts

```diff
@@ -168,15 +168,18 @@ export class DovetailAudio extends ExtendableAudio {
     this._dovetailLoading = true;
     let promise = this.currentPromise = this.dovetailFetcher.fetch(url).then(
       result => {
-        if (this.currentPromise === promise) {
+        if (!result.request.placements.length) {
+          throw new Error('No placements');
+        } else if (this.currentPromise === promise) {
           let data: AllUnion  = [result, this.getArrangement(result.request)];
           return Promise.all<any>(data);
         }
       }
     ).then(([dovetailResponse, adzerkResponse]: AllResult) =>  {
       return this.calculateArrangement(dovetailResponse.arrangement, adzerkResponse);
     }).catch(error => {
-      return this.fallback(url);
+      let directUrl = this.dovetailFetcher.transform(url);
```

Comment:
Nitpick: there's an off chance that we mis-recognized a dovetail URL, and `dovetailFetcher.fetch()` got an actual mp3 file.  That would throw an error (cannot json-decode an mp3 file), and end up here.

So probably safer for this `.catch(` to continue redirecting to the original URL.